core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -937,4 +938,54 @@ default HttpResponse recover(Function<? super Throwable, ? extends HttpResponse>
         requireNonNull(function, "function");
         return of(new RecoverableStreamMessage<>(this, function, /* allowResuming */ false));
     }
+
+    /**
+     * Recovers a failed {@link HttpResponse} by switching to a returned fallback {@link HttpResponse}
+     * when the thrown {@link Throwable} is the same type or a subtype of the
+     * specified {@code causeClass}.
+     *
+     * <p>Example:<pre>{@code
```

Comment:
Should I add that example to `recover(Function<? super Throwable, ? extends HttpResponse>)` method as well?
It doesn't seem like the correct way to use it, but it can be used.

Reply:
I think it's enough for now. Thanks for updating the Javadoc 🙇